### PR TITLE
Changelogs for 7.42.0 release

### PR DESCRIPTION
**Pull request by kacper-murzyn**

### CODE REVIEW

**Review by amenasria — APPROVED** [review]
LGTM for Agent Platform

File: CHANGELOG.rst [hunk 1]
```diff
@@ -2,6 +2,191 @@
...
+
+
+.. _Release Notes_7.42.0_Upgrade Notes:
+
+Upgrade Notes
+-------------
+
+- Downloading and installing official checks with `agent integration install`
+  is no longer supported for Agent installations that do not include an embedded
+  python3.
```
Comment (olivielpeau) [inline CHANGELOG.rst:27]: (not blocking for PR merge, we can address later) we should clarify which Agent deployments this actually affects in terms of agent version/configuration

File: CHANGELOG.rst [hunk 2]
```diff
@@ -2,6 +2,191 @@
...
+
+- Starting with Agent 7.40, the Agent will fail to start when it is unable to determine the hostname, instead of silently using an irrelevant hostname (usually, a container ID).
+  Hostname resolution is key to many features and failure to determine hostname means that the Agent is not configured properly.
+  This change mostly affects Agents running in containerized environments as we cannot rely on the OS hostname.
```
Comment (olivielpeau) [inline CHANGELOG.rst:31]: Let's delete this, it's not meant to be added to 7.42's changelog (see https://github.com/DataDog/datadog-agent/pull/14249/)

**Review by olivielpeau — COMMENTED** [review]
LGTM overall, just 2 comments (no need for me to re-review before merging :) )

File: CHANGELOG.rst [hunk 3]
```diff
@@ -2,6 +2,191 @@
...
+- Adding a configuration option, ``exclude_ec2_tags``, to exclude EC2 instance tags from being converted into host
+  tags.
+
+- Adds detection for a process being executed directly from memory without the binary present on disk.
+
+- Introducing agent sampling rates remote configuration.
+
+- Expose process parent in SECL
+
+- Adds support for ``secret_backend_command_sha256`` SHA for the ``secret_backend_command`` executable. If ``secret_backend_command_sha256`` is used,
```
Comment (clarkb7) [inline CHANGELOG.rst:57]: @xornivore This PR was reverted right?

Comment (xornivore) [reply CHANGELOG.rst:57]: Yup, correct, @kacper-murzyn should we remove it?

Comment (xornivore) [reply CHANGELOG.rst:57]: Thank you @clarkb7 ! 🙌

Comment (kacper-murzyn) [reply CHANGELOG.rst:57]: In this case yes, I can do that. Thanks for checking

File: CHANGELOG.rst (hunk 2 above)
Comment (kacper-murzyn) [reply CHANGELOG.rst]: Removed

**Pull request merged**